app/jobs/track_agent_statuses_job.rb

```diff
@@ -8,10 +8,34 @@ def perform
                       status: data[:status],
                       time_in_status: data[:time_in_status])
     end
+    lunch current
     log = JobLog.new('TrackAgentStatusesJob')
     AgentStatusUpdater.new(now, log.last_success).update_statuses(current)
   end
 
+  def lunch(states)
+    luncheds = Rails.cache.read 'lunched'
+
+    @contacts_service = ContactsService.new
+    time = Time.zone.now
+    start_time = time.beginning_of_day
+    team_name = 'Helpdesk'
+
+    if luncheds.nil?
+      luncheds = Set.new
+      eaters = @contacts_service.statuses(team_name, start_time, time, 'Ruokatunti')
+      eaters.each do |eater|
+        luncheds.add eater.id
```

Comment:
This is wrong because it stores id of the status instead of the agent. The correct behaviour should be tested.

Also simple optimization/simplification would be to use `eaters.plunk(:agent_id)` that fetches agent ids from database as array.